pkg/controller/pxc/upgrade.go

```diff
@@ -79,8 +79,9 @@ func (r *ReconcilePerconaXtraDBCluster) updatePod(sfs api.StatefulApp, podSpec *
 		if err != nil {
 			return fmt.Errorf("pmm container error: %v", err)
 		}
-
-		newContainers = append(newContainers, pmmC)
+		if pmmC.Image != "" {
```

Comment:
I belive we must check if PMM is enabled in general
```suggestion
		if cr.Spec.PMM != nil && cr.Spec.PMM.Enabled {
```

Reply:
@delgod pmm is enabled or not is not the case here as we don't need for haproxy in either way as it is enabled for the other parts like pxc or proxysql

Reply:
Image can be passed from CR or from Version Service, it is not reliable, I have reworked on `pmmC.Name`